high_card.rb

```diff
@@ -3,3 +3,42 @@
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
 players = []
+deck = []
+winners = []
+
+ranks.each do |rank|
+  suits.each do |suit|
+    deck.push([rank, suit])
+  end
+end
+
+deck.shuffle!
+
+while true
+  puts "#{players.length} players so far. Enter a player name, or type 'play':"
+  player_name = gets.chomp
+  break if player_name == "play"
+  players.push(player_name)
+end
+
+dealt = players.map do |player|
+  deck.pop
+end
+
+dealt_value = dealt.map do |dealt|
+  if dealt[0][0].is_a?(String) then
+    dealt[0][0].gsub(/[JKQA]/, 'J' => 11, 'Q' => 12, 'K' => 13, 'A' => 14).to_i
```

Comment:
Regex, very cool! Encourage you to think about non-Regex ways you could address this problem (e.g., `each_with_index`).